code/modules/shareddream/dream_entry.dm

```diff
@@ -29,18 +48,36 @@ var/list/dream_entries = list()
 		if(istype(bg) || force_wakeup)
 			// If we choose to be asleep, keep sleeping.
 			if(willfully_sleeping && sleeping && stat == UNCONSCIOUS)
-				sleeping = 5
-				return
+				if(can_commune() || (!can_commune() && srom_pulled_by))
```

Comment:
```suggestion
				if(can_commune() || srom_pulled_by)
```

Logically we only get to the second part if can_commune is false, don't need to check it again.